firebase-common/src/main/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatInfo.java

```diff
@@ -18,22 +18,57 @@
 import androidx.annotation.NonNull;
 import androidx.annotation.RestrictTo;
 import androidx.annotation.VisibleForTesting;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.TaskCompletionSource;
 import com.google.firebase.components.Component;
 import com.google.firebase.components.Dependency;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
 
 /** Provides information as whether to send heart beat or not. */
 public class DefaultHeartBeatInfo implements HeartBeatInfo {
 
   private HeartBeatInfoStorage storage;
 
-  private DefaultHeartBeatInfo(Context context) {
+  private Set<HeartBeatConsumer> consumers;
+
+  private ExecutorService backgroundExecutor;
+
+  private static final ThreadFactory THREAD_FACTORY =
+      new ThreadFactory() {
+        private final AtomicInteger mCount = new AtomicInteger(1);
+
+        @Override
+        public Thread newThread(Runnable r) {
+          return new Thread(
+              r, String.format("heartbeat-information-executor-%d", mCount.getAndIncrement()));
+        }
+      };
+
+  private DefaultHeartBeatInfo(Context context, Set<HeartBeatConsumer> consumers) {
```

Comment:
pls have this constructor delegate to the one below

Reply:
hmm so the other constructor is test only and can't be accessed via this

Reply:
in this case pls change the annotation to `@VisibleForTesting` so that linter does not complain